internal/pkg/call/call.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package call contains functionality for determining whether a function
+// call refers to a given referrer.
+package call
+
+import (
+	"golang.org/x/tools/go/ssa"
+)
+
+// Call is a wrapper for *ssa.Call instructions, allowing Variadic and
+// non-Variadic calls to be handled polymorphically
+type Call interface {
```

Comment:
Future improvements:
I am curios what it would look like if we were to combine `varargs` pkg (which is really a call) with this one.
In other words, this package would hide the complexity involved in dealing with regular calls vs. variadic.

Reply:
Agreed. In fact, I think this new `Call` concept could become a whole `Sink` type, possibly produced by a separate `Analyzer`.